File: src/main/java/com/github/jesty/reactiveredis/TickerApiDelegateImpl.java
package com.github.jesty.reactiveredis;

import java.math.BigDecimal;
import java.time.Duration;

import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ServerWebExchange;

import com.github.jesty.reactiveredis.web.api.TickerApiDelegate;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class TickerApiDelegateImpl implements TickerApiDelegate {

	@Override
	public Mono<ResponseEntity<Flux<BigDecimal>>> tick(ServerWebExchange exchange) {
		Flux<BigDecimal> myflux = exchange	.getRequest()
											.getHeaders()
											.getAccept()
											.stream()
											.filter(mediaType -> mediaType.equals(MediaType.TEXT_EVENT_STREAM))
											.findFirst()
											.map(mt -> buildFlux())
											.orElse(Flux.just(BigDecimal.ONE));

		return Mono.just(ResponseEntity	.ok()
										.body(myflux));

	}

	private Flux<BigDecimal> buildFlux() {
		return Flux	.range(1, 10)
					.map(n -> new BigDecimal(n))
					.doOnNext(System.out::println)
					.delayElements(Duration.ofSeconds(1));
	}

}

File: src/main/kotlin/com/github/jesty/webfluxreactive/TickerApiDelegateImpl.kt
package com.github.jesty.webfluxreactive

import com.github.jesty.webfluxreactive.api.TickerApiDelegate
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import org.springframework.web.server.ServerWebExchange
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.math.BigDecimal
import java.time.Duration.ofSeconds

@Service
class TickerApiDelegateImpl : TickerApiDelegate {

    override fun tick(exchange: ServerWebExchange?): Mono<ResponseEntity<Flux<BigDecimal>>> {
        val flux = Flux.range(1, 10)
                .map { BigDecimal(it) }
                .doOnNext { println(it) }
                .delayElements(ofSeconds(1))
        return Mono
                .just(ResponseEntity
                        .ok()
                        .body(flux))
    }

}